ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -639,14 +644,31 @@ private ReplicaMetadataResponse getReplicaMetadataResponse(List<RemoteReplicaInf
   }
 
   /**
-   * Gets the missing store messages by comparing the messages from the remote node
+   * Determines if {@link MessageInfo} container in the status of DELETED_IN_PROGRESS or INACTIVE.
+   * @param messageInfo A message info class that contains basic info about a blob
+   * @return {@code true} if the blob associates with the deprecated container, {@code false} otherwise.
+   * Deprecated containers status include DELETE_IN_PROGRESS and INACTIVE.
+   */
+  private boolean isDeprecatedContainer(MessageInfo messageInfo) {
+    if (accountService != null) {
+      Container.ContainerStatus status = accountService.getAccountById(messageInfo.getAccountId())
+          .getContainerById(messageInfo.getContainerId())
+          .getStatus();
+      return status == Container.ContainerStatus.DELETE_IN_PROGRESS || status == Container.ContainerStatus.INACTIVE;
```

Comment:
Should replication be subject to the same retention time after marked as DELETE_IN_PROGRESS that the compactor is?

Reply:
Once the container marked as DELETE_IN_PROGRESS, the ACLs will be removed and user can't access this container. In the mean time, should we skip the associate blobs in replication directly? Unless for the future we want to undo the container deletion within the retention time, in that situation we will need to involve retention time for replication. What do you think?

Reply:
I guess I was thinking that not replicating the blobs is about the same as compacting them away in that it would be very hard to revert this decision. Without erasing the replica tokens, replication can't go back in time to get the missing blobs unless a new event for the blob appears on a peer like a TTLUPDATE or UNDELETE.

What was the motivation for the retention time within compaction? My understanding was that it was so that the container deletion could be reverted if it was done by mistake. Sorry if that was a bad assumption on my part.

Reply:
The retention time what I add previously is due to after ACLs has been removed we need some retention time before we start compaction. But you are right, I need to consider about revert the container deletion as well. So I add the retention time same as what we are doing in compaction, so that after within the retention time, the DELETE_IN_PROGRESS container won't be skipped through replication. And it's easier for revert as you mentioned.